contracts/implementations/meta/MetaUSD.vy

```diff
@@ -800,12 +834,22 @@ def exchange_underlying(
     else:
         # If both are from the base pool
         dy = ERC20(output_coin).balanceOf(self)
-        Curve(base_pool).exchange(base_i, base_j, dx_w_fee, _min_dy)
+        Curve(base_pool).exchange(base_i, base_j, dx, _min_dy)
         dy = ERC20(output_coin).balanceOf(self) - dy
 
-    ERC20(output_coin).transfer(_receiver, dy)
+    response = raw_call(
+        output_coin,
+        concat(
+            method_id("transfer(address,uint256)"),
+            convert(_receiver, bytes32),
+            convert(dy, bytes32),
+        ),
+        max_outsize=32,
+    )
+    if len(response) > 0:
+        assert convert(response, bool)
 
-    log TokenExchangeUnderlying(msg.sender, i, _dx, j, dy)
+    log TokenExchangeUnderlying(msg.sender, i, dx, j, dy)
```

Comment:
_dx is not equal to dx when coin_i is from base_pool, in this case, _dx is the number of coin_i, but dx is the amount of 3crv from which coin_i deposit into base_pool.